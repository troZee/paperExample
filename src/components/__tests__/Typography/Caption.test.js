import * as React from 'react';
import renderer from 'react-test-renderer';
import Caption from '../../Typography/Caption.tsx';

it('renders caption with children as content', () => {
  const tree = renderer.create(<Caption>Caption content</Caption>).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders caption applying style', () => {
  const tree = renderer
    .create(
      <Caption style={{ fontSize: 20, color: 'red' }}>
        Big and red caption
      </Caption>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
