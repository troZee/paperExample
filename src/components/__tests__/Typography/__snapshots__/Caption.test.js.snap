// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders caption applying style 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#000000",
        "fontFamily": "System",
        "fontWeight": "400",
        "textAlign": "left",
      },
      Array [
        Object {
          "color": "rgba(0, 0, 0, 0.54)",
          "fontFamily": "System",
          "fontWeight": "400",
          "textAlign": "left",
          "writingDirection": "ltr",
        },
        Array [
          Object {
            "fontSize": 12,
            "letterSpacing": 0.4,
            "lineHeight": 20,
            "marginVertical": 2,
          },
          Object {
            "color": "red",
            "fontSize": 20,
          },
        ],
        Array [
          Object {
            "fontSize": 12,
            "letterSpacing": 0.4,
            "lineHeight": 20,
            "marginVertical": 2,
          },
          Object {
            "color": "red",
            "fontSize": 20,
          },
        ],
      ],
    ]
  }
>
  Big and red caption
</Text>
`;

exports[`renders caption with children as content 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#000000",
        "fontFamily": "System",
        "fontWeight": "400",
        "textAlign": "left",
      },
      Array [
        Object {
          "color": "rgba(0, 0, 0, 0.54)",
          "fontFamily": "System",
          "fontWeight": "400",
          "textAlign": "left",
          "writingDirection": "ltr",
        },
        Array [
          Object {
            "fontSize": 12,
            "letterSpacing": 0.4,
            "lineHeight": 20,
            "marginVertical": 2,
          },
          undefined,
        ],
        Array [
          Object {
            "fontSize": 12,
            "letterSpacing": 0.4,
            "lineHeight": 20,
            "marginVertical": 2,
          },
          undefined,
        ],
      ],
    ]
  }
>
  Caption content
</Text>
`;
